hooks/brc20/useBrc20TransferExecute.ts

```diff
@@ -0,0 +1,130 @@
+import { useEffect, useState } from 'react';
+
+import { NetworkType, UTXO } from 'types';
+import { ExecuteTransferProgressCodes, brc20TransferExecute } from '../../transactions/brc20';
+
+export enum ErrorCode {
+  INSUFFICIENT_FUNDS = 'INSUFFICIENT_FUNDS',
+  INVALID_TICK = 'INVALID_TICK',
+  INVALID_AMOUNT = 'INVALID_AMOUNT',
+  INVALID_FEE_RATE = 'INVALID_FEE_RATE',
+  TRANSFER_BROADCAST_FAILED = 'TRANSFER_BROADCAST_FAILED',
+  SERVER_ERROR = 'SERVER_ERROR',
+}
+
+/**
+ *
+ * @param seedPhrase - The seed phrase of the wallet
+ * @param accountIndex - The account index of the seed phrase to use
+ * @param addressUtxos - The UTXOs in the bitcoin address which will be used for payment
+ * @param tick - The 4 letter BRC-20 token name
+ * @param amount - The amount of the BRC-20 token to transfer
+ * @param revealAddress - The address where the balance of the BRC-20 token lives. This is usually the ordinals address.
+ * @param changeAddress - The address where change SATS will be sent to. Should be the Bitcoin address of the wallet.
+ * @param recipientAddress - The address where the BRC-20 tokens will be sent to.
+ * @param feeRate - The desired fee rate for the transactions
+ * @param network - The network to broadcast the transactions on (Mainnet or Testnet)
+ * @returns
+ */
+const useBrc20TransferExecute = (
+  seedPhrase: string,
+  accountIndex: number,
+  addressUtxos: UTXO[],
+  tick: string,
+  amount: number,
+  revealAddress: string,
+  changeAddress: string,
+  recipientAddress: string,
+  feeRate: number,
+  network: NetworkType,
+) => {
+  const [executed, setExecuted] = useState(false);
+  const [running, setRunning] = useState(false);
+  const [complete, setComplete] = useState(false);
+  const [transferTransactionId, setTransferTransactionId] = useState<string | undefined>();
+  const [progress, setProgress] = useState<ExecuteTransferProgressCodes | undefined>();
+  const [errorCode, setErrorCode] = useState<ErrorCode | undefined>();
+
+  const executeTransfer = () => {
+    setExecuted(true);
+  };
+
+  useEffect(() => {
```

Comment:
consider not using a useEffect here and just export method which is charge of executing, like executeTransfer. for example

```ts
const executeTransfer = () => {
const {errorCode} = validateParams()
if(errorCode) {
return setErrorCode(errorCode); 
}
try {
        const transferGenerator = await brc20TransferExecute({
...
}
```
that way you don't need executed state